contracts/RelayEncoder.sol

```diff
@@ -0,0 +1,128 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+contract RelayEncoder { // NOTE: this is logic for proxy contract
+
+    // dev Encode 'bond' relay call
+    // @param controller_address: Address of the controller
+    // @param amount: The amount to bond
+    // @param reward_destination: the account that should receive the reward
+    // @returns The bytes associated with the encoded call
+    function encode_bond(uint256 controller_address, uint256 amount, bytes memory reward_destination) external pure returns (bytes memory result) {
+        return bytes.concat(hex"0600", hex"00", bytes32(controller_address), scaleCompactUint(amount), reward_destination);
+    }
+
+    // dev Encode 'bond_extra' relay call
+    // @param amount: The extra amount to bond
+    // @returns The bytes associated with the encoded call
+    function encode_bond_extra(uint256 amount) external pure returns (bytes memory) {
+        return bytes.concat(hex"0601", scaleCompactUint(amount));
+    }
+
+    // dev Encode 'unbond' relay call
+    // @param amount: The amount to unbond
+    // @returns The bytes associated with the encoded call
+    function encode_unbond(uint256 amount) external pure returns (bytes memory) {
+        return bytes.concat(hex"0602", scaleCompactUint(amount));
+    }
+
+    // dev Encode 'rebond' relay call
+    // @param amount: The amount to rebond
+    // @returns The bytes associated with the encoded call
+    function encode_rebond(uint256 amount) external pure returns (bytes memory) {
+        return bytes.concat(hex"0613", scaleCompactUint(amount));
+    }
+
+    // dev Encode 'withdraw_unbonded' relay call
+    // @param slashes: Weight hint, number of slashing spans
+    // @returns The bytes associated with the encoded call
+    function encode_withdraw_unbonded(uint32 slashes) external pure returns (bytes memory) {
+        if (slashes < 1<<8) {
+            return bytes.concat(hex"0603", bytes1(uint8(slashes)), bytes3(0));
+        }
+        if(slashes < 1 << 16) {
+            uint32 bt2 = slashes / 256;
```

Comment:
bitwise shift should be lighter ?  slashes << 8

Reply:
Not Fixed